### ISSUE TITLE
Stop `search_data_dict` from returning an Error when nothing found

### ISSUE THREAD

**Issue by seabbs** [seabbs/getTBinR#65]
**Is your feature request related to a problem? Please describe.**
When `search_data_dict` finds nothing currently an error is returned. This is not ideal as it can cause downstream functions to fail if a variable has no entry in the data dictionary. The default should be to fail but return the base metric as the label in these cases. 

**Describe the solution you'd like**
Change `search_data_dict` to return something other than an error when nothing is found. 

**Describe alternatives you've considered**
Change all use cases to cope better when `search_data_dict` fails

**Additional context**
Reproduce with the following: 

```
library(getTBinR)
search_data_dict('g4gt23q2')
```

**Comment by seabbs** [comment]
Updated `search_data_dict` to return `NULL` when nothing is found. 

Example now gives:

```
Loading data from: /tmp/RtmpLF0WAL/TB_data_dict.rds
No results found for your variable search for g4gt23q2
NULL
```

**Comment by seabbs** [comment]
Updated tests of `search_data_dict` to check this.

**Comment by seabbs** [comment]
Updated`prepare_df_plot` to properly handle custom metrics that are not in the data dictionary.

**Comment by seabbs** [comment]
Fixed by #7ec01f6
